docs/plugins/repository-s3.asciidoc

```diff
@@ -236,6 +236,16 @@ The following settings are supported:
     currently supported by the plugin. For more information about the
     different classes, see http://docs.aws.amazon.com/AmazonS3/latest/dev/storage-class-intro.html[AWS Storage Classes Guide]
 
+`client_symmetric_key`::
+    Sets the keys to use to encrypt your snapshots. You can specify either a symmetric key or a public/private key pair.
+    No encryption by default. This sets a Base64-encoded AES symmetric-key (128, 192 or 256 bits)
+
+`client_public_key`::
+    Sets the a base64-encoded RSA public key
+
+`client_private_key`::
+    Sets the a base64-encoded RSA private key
+
```

Comment:
I was wondering about this. It's all about encryption.

We already have `server_side_encryption`. I wonder if we should at some point rename the settings (may be within another PR) to:

```js
PUT _snapshot/my_s3_repository
{
  "type": "s3",
  "settings": {
    "bucket": "my_bucket_name",
    "region": "us-west",
    "encryption": {
       "server": {
           "enabled":  false
       },
       "client": {
           "enabled":  true,
           "symmetric_key": "XYZ",
           "public_key": "XYZ",
           "private_key": "XYZ"
       }
    }
  }
}
```

Just thinking out loud here. Might be too much engineering though...